server/src/main/java/org/apache/druid/server/ClientQuerySegmentWalker.java

```diff
@@ -607,7 +607,7 @@ private <T> QueryRunner<T> decorateClusterRunner(Query<T> query, QueryRunner<T>
    * @param parentQueryResourceId Parent Query's Query Resource ID
    * @return DataSource populated with the subqueries
    */
-  private DataSource generateSubqueryIds(
+  public static DataSource generateSubqueryIds(
```

Comment:
Why public?

Reply:
The CachingClusteredClient benchmark was broken as subquery Ids were not present in the queries it was running. Let me annotate it with a "visible for testing" for clarity.